### PR TITLE
#162948908 Receive descriptive validation error message

### PR DESCRIPTION
**Pull request by jamesbeamie**
#### What does this PR do?

This PR has validation for user registration and login, returning descriptive validation error messages.
#### Description of the task to be accomplished?

* Create tests for validation
* Create validation function for username, email, and password

#### How can this be manually tested?

* Clone this repository
* CD into the project directory
* Install dependencies `pip install -r requirements.txt`
* Run this command to run tests `python3 manage.py test`
* Run this command to start the server `python3 manage.py startserver`
* Register a user with special characters included in the username
* Register a user with password less than 8 characters.

#### Background context?

* Use Python3 version `3.7 or above 3.x`
* Use Django version `2.x`

#### Related Pivotal Tracker stories?

* [#162948908](https://www.pivotaltracker.com/story/show/162948908)

### CODE REVIEW

**Review by danuluma — APPROVED** [review]

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -6,70 +6,69 @@
...
+	LoginSerializer, RegistrationSerializer, UserSerializer
 )
 
 
-class RegistrationAPIView(GenericAPIView):
-    # Allow any user (authenticated or not) to hit this endpoint.
-    permission_classes = (AllowAny,)
-    renderer_classes = (UserJSONRenderer,)
-    serializer_class = RegistrationSerializer
+class RegistrationAPIView(APIView):
```
Comment (frankip) [inline authors/apps/authentication/views.py]: @jamesbeamie what's the reason for changing from `GenericAPIView`

**Review by frankip — COMMENTED** [review]
@jamesbeamie good work in your implementations however in the spirit of DRY and abstraction this can still be improved. For instance you can create a new file called maybe `validators` which will house all the validation methods. That goes the same for the error messages. create a file called something like `messages` which will contain dictionaries of the error message.

```
success = {
'user_registered': 'you registered successful`
}
errors= {
'bad_email': 'Please input a valid email.`
}
```

so that when you want to show a response message 
you import the file and call something like errors['bad_email']
This simplifies and reduces the need to repeat your self. this can also be used when asserting the tests.
You can always reach out if it doesn't make sense right away.

File: authors/apps/authentication/views.py (hunk 1 above)
Comment (jamesbeamie) [reply authors/apps/authentication/views.py]: This changed because I had not rebased from develop

Comment (jamesbeamie) [reply authors/apps/authentication/views.py]: I am rebasing to fix this

**Comment by boltonotieno** [comment]
You could change the PR title to
```#162948908 Receive descriptive validation error message```

File: authors/apps/authentication/messages.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "missmatch":"Email and password did not match",
+    "require_pwd":"Password is required to login",
+    "email_pwd":"Email address is required to login",
+}
+
+success_msg={
+    "confirmed_email":"Email succesfully confirmed",
+    "pwd_changed":"Your password was successfully changed",
+    "request_success":"Request for password reset was succesful"
+}
```
Comment (frankip) [inline authors/apps/authentication/messages.py]: add a new line here

**Review by boltonotieno — APPROVED** [review]

**Review by Deekerubo — APPROVED** [review]

**Review by Ruiru11 — APPROVED** [review]

**Review by frankip — APPROVED** [review]
LGTM

**Pull request merged**